### PR TITLE
Make output file names unique when rendering videos in documentation

### PR DESCRIPTION
**Pull request by behackl**
A quick follow-up to #402.

## List of Changes
- Allow duplicate scene names for examples rendered by the manim directive.

## Motivation
Scenes with the same name are currently saved with the same name. Avoid this so that people can use `Example(Scene)` however often they wish. :-)

## Explanation for Changes
This adds a dict `classnamedict` keeping track of the already rendered scenes. With it, a number is added to the output file name to make it unique.

## Testing Status
Works locally as intended; RTD should also be compatible. 

## Acknowledgement
- [x] I have read the [Contributing Guidelines](https://github.com/ManimCommunity/manim/wiki/Documentation-guidelines-(WIP))

### CODE REVIEW

**Review by kolibril13 — APPROVED** [review]

**Comment by kolibril13** [comment]
Looks good to me, and a very convenient feature!
I did not test it, but I think if it works on your machine, it will also work on others.
One question that comes to my mind:
Will this also work in case of class names:
`Example, Example, Example1 , Example2`
Won't the second `Example` then collide with Example1?

File: docs/source/manim_directive.py [hunk 1]
```diff
@@ -147,6 +155,7 @@ def run(self):
         media_dir = os.path.join("source", "media")
         images_dir = os.path.join(media_dir, "images")
         video_dir = os.path.join(media_dir, "videos")
+        output_file = f"{clsname}-{classnamedict[clsname]}"
```
Comment (kolibril13) [inline docs/source/manim_directive.py:158]: Maybe we can set here something more unambiguous like :
output_file = f"{clsname}-occurrence{classnamedict[clsname]}"

**Review by kolibril13 — APPROVED** [review]

File: docs/source/manim_directive.py (hunk 1 above)
Comment (behackl) [reply docs/source/manim_directive.py:158]: Sure, any modification will work.

Given that nobody is ever supposed to interact with these files (except for the docbuild system, which is pretty verbose about what it is currenly processing anyways), I went with a very minimal version. But I'm happy to be more explicit. :-)

**Comment by behackl** [comment]
> One question that comes to my mind:
> Will this also work in case of class names:
> `Example, Example, Example1 , Example2`
> Won't the second `Example` then collide with Example1?

No, this cannot happen as the name of every scene is modified: in your example, the output names would be `Example-1`, `Example-2`, `Example1-1` and `Example2-1`.

**Review by leotrs — APPROVED** [review]

**Pull request merged**